control/obstacle_collision_checker/src/obstacle_collision_checker_node/obstacle_collision_checker_node.cpp

```diff
@@ -28,16 +28,21 @@
 
 namespace
 {
-template <typename T>
-void update_param(
-  const std::vector<rclcpp::Parameter> & parameters, const std::string & name, T & value)
+template <class T>
+bool update_param(
+  const std::vector<rclcpp::Parameter> & params, const std::string & name, T & value)
 {
-  auto it = std::find_if(
-    parameters.cbegin(), parameters.cend(),
-    [&name](const rclcpp::Parameter & parameter) { return parameter.get_name() == name; });
-  if (it != parameters.cend()) {
-    value = it->template get_value<T>();
+  const auto itr = std::find_if(
```

Comment:
@brkay54 Since it seems that bug-fix and refactoring are mixed, could you split PRs into the appropriate sizes?

Reply:
Thank you for your review. I am going to close this PR and create 2 PRs for bug-fix and refactoring.